### ISSUE TITLE
nuxt/kit: ability to configure web components for auto-import

### ISSUE THREAD

**Issue by WickyNilliams** [nuxt/nuxt#25528]
### Describe the feature

I have been trying to configure auto-imports for a library of web components, so that i can minimal bundle size without having to manually add imports.

I tried creating a module like this:

```ts
import { defineNuxtModule, addComponent } from 'nuxt/kit'

export default defineNuxtModule({
  meta: {
    name: 'my-components',
    compatibility: {
      nuxt: '^3.0.0',
    },
  },

  setup() {
    // From the runtime directory
    addComponent({
      name: 'MyButton',
      kebabName: 'my-button',
      mode: 'client',
      filePath: "@my/web-components/Button.js",
    })
  },
})
```

However, this does not work as `addComponent` (or something downstream of it) expects there to be a vue component at the given path. the problem would exist just the same if I had a web component relative to the module (i.e. not from npm)

I am wondering if there is any scope to extend `addComponent`/component auto-import behavior to support web components. Since web components typically register themselves, the desired output would be a side-effect import wherever usage of `my-button` was detected. something like this:

```ts
// rather than this
import Button from  "@my/web-components/Button.js"
// we would have this
import "@my/web-components/Button.js"
```

My first thought was to add a `sideEffectOnly: boolean` option to `addComponent` as a flag to indicate this is the desired behavior, but i don't think that composes well with other existing options (e.g. what does it mean to have `mode: "server"` with `sideEffectOnly: true`?). Therefore perhaps the best option would be a new value for `mode`, since (right now) web components are implicitly client-side. 

I'm sure it's possible to build this behavior outside of nuxt with some custom transforms etc, but i imagine a comprehensive solution would end up recreating most, if not all, of nuxt's existing mechanisms for scanning/importing/tree-shaking components.

Such an addition would be useful not just in my use case, but for all web component libraries. Additionally, many WC libraries publish a [manifest of components](https://github.com/webcomponents/custom-elements-manifest) they contain,  meaning a generic nuxt module could be written which accepts the manifest as an input, and dynamically calls `addComponent` for each web component found. 

In summary, my proposal is something like:

```ts
addComponent({
  name: 'my-button',
  mode: 'web-component', // the specific value is up for debate
  filePath: "@my/web-components/Button.js",
})
```

### Additional information

- [X] Would you be willing to help implement this feature?
- [X] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by danielroe** [comment]
Would welcome thoughts from @prashantpalikhe and @steveworkman.

**Comment by steveworkman** [comment]
I agree that adding imports manually is a chore, and in a CSR Vue app it's the only way to split the WC imports effectively due to their inherent side effects. That given, it does work very well.

Assuming that a WC library had a manifest (not all of them do) and the template parser could pick up WCs by their tags as defined in the next config, then it is plausible that they could be auto-imported.

If there is no manifest then I don't see how this can work. One could be supplied in the next config but at that point you may as well have put the imports in the right components.

Thinking of this in relation to the SSR-lit plugin, we had issues with the import order of some of the lit dependencies. The WCs that are detected to be imported would need to be included along with other route/component specific imports so that they are loaded late enough for the SSR plugins to be ready for them.

Happy to test this out if it gets a PR.

**Comment by WickyNilliams** [comment]
Thanks Steve. Appreicate your perspective especially with regards to SSR. I would personally, as a first step, want to keep the scope very narrow so that it only covers client side rendering, and _only_ extends nuxt kit to allow auto imports of web components. Consuming a manifest, adding options to nuxt config, or SSR concerns can be added later/independently. 

> If there is no manifest then I don't see how this can work. One could be supplied in the next config but at that point you may as well have put the imports in the right components

I'm not sure I agree. The nature of WCs mean that there is no identifier to import. So you get no type errors at build time if you fail to import a WC. And vscode offers no suggestion to add an import. So an import is very easily forgotten. And when you do forget, you get (often silent) failures at runtime. 

Worse yet, because web components are in a global registry, you may not get any failures even if you forget an import. Consider `ComponentA` and `ComponentB` both depend on a web component `my-button`. `ComponentA` correctly imports `my-button`, whilst `ComponentB` did not. 

In this scenario, `ComponentB` may render perfectly fine with no errors, simply because `ComponentA` was also loaded on the current (or a previous) page, meaning `my-button` is already in the global registry. `ComponentB` may work only by coincidence, despite its missing imports. This can get confusing and hard to track down, especially as the module graph gets more complex. 

So there is still value even if manually defining each web component for auto-import. Which is why I think extending nuxt kit _just enough_ is useful in itself, and it can act as an enabler for better/smarter functionality to be built on top later.
